src/components/avatar/avatar.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { ComponentType } from 'react';
+
+export type AvatarProps = {
+	size: 's' | 'm' | 'l' | 'xl';
+	ImageComponent?: ComponentType<{ className: string }>;
```

Comment:
Instead of using the classname of as an identifier for the component type, I would rather change that to src: string and alt: string. Every image, be it next or default html, has these two attributes. className is also a bit to generic.

Reply:
@malinovic 

The className is used to write classes directly on the image component in the render function:
 <ImageComponent className={imageClasses} />

The implemented Solution is now that I replaced the prop definition with: ComponentType<{ src: string; alt: string }>. As a consequence in the render function I have to write: 

			<div className={imageClasses}>
				{ImageComponent ? <ImageComponent src="src" alt="User Avatar" /> : image}
			</div>


A third and maybe preferable solution would be to just define the prop as ComponentType<> and in the render function do: 

			<div className={imageClasses}>
				{ImageComponent ? <ImageComponent/> : image}
			</div>